react/features/authentication/components/LoginDialog.native.js

```diff
@@ -0,0 +1,271 @@
+ * the {@link JitsiConference} instance. If user decides to authenticate a new
+ * {@link JitsiAuthConnection} will be created from which separate XMPP
+ * connection is established and authentication is performed. In case it
+ * succeeds Jicofo will assign new session ID which then can be used from
+ * the anonymous domain connection to create and join the room. This part is
+ * done by {@link JitsiAuthConnection} from lib-jitsi-meet.
+ *
+ * See https://github.com/jitsi/jicofo#secure-domain for configuration
+ * parameters description.
+ */
+class LoginDialog extends Component {
+    /**
+     * LoginDialog component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * {@link JitsiConference} that needs authentication - will hold a valid
+         * value in XMPP login + guest access mode.
+         */
+        _conference: PropTypes.object,
+
+        /**
+         *
+         */
+        _configHosts: PropTypes.object,
+
+        /**
+         * Indicates if the dialog should display "connecting" status message.
+         */
+        _connecting: PropTypes.bool,
+
+        /**
+         * The error which occurred during login/authentication.
+         */
+        _error: PropTypes.string,
+
+        /**
+         * Any extra details about the error provided by lib-jitsi-meet.
+         */
+        _errorDetails: PropTypes.string,
+
+        /**
+         * Redux store dispatch method.
+         */
+        dispatch: PropTypes.func,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: PropTypes.func
+    };
+
+    /**
+     * Initializes a new LoginDialog instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            username: '',
+            password: ''
+        };
+
+        // Bind event handlers so they are only bound once per instance.
+        this._onCancel = this._onCancel.bind(this);
+        this._onLogin = this._onLogin.bind(this);
+        this._onPasswordChange = this._onPasswordChange.bind(this);
+        this._onUsernameChange = this._onUsernameChange.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const {
+            _connecting: connecting,
+            _error: error,
+            _errorDetails: errorDetails,
+            t
+        } = this.props;
+
+        let messageKey = '';
+        const messageOptions = {};
+
+        if (error === JitsiConnectionErrors.PASSWORD_REQUIRED) {
+            messageKey = 'dialog.incorrectPassword';
+        } else if (error) {
+            messageKey = 'dialog.connectErrorWithMsg';
+            messageOptions.msg = `${error} ${errorDetails}`;
+        }
+
+        return (
+            <Dialog
```

Comment:
After I switched from `Modal` to `Dialog`, we need titles for `LoginDialog` and `WaitForOwnerDialog` please.

Reply:
I took care of the titles. I used the same titles as Web (and I made the `LoginDialog` look very similar to Web's one).